sdk/python/lib/pulumi/provider/experimental/analyzer.py

```diff
@@ -401,10 +422,75 @@ def analyze_property(
                 ref=ref,
                 optional=optional,
                 plain=plain,
+                description=self.get_docstring(typ.__name__, name),
             )
         else:
             raise ValueError(f"unsupported type {arg}")
 
+    def find_docstrings(self, path: Path) -> dict[str, dict[str, str]]:
+        """
+        find_docstrings returns the docstrings for all the attributes of all
+        the classes in `self.path.
+
+        Unfortunately, only class docstrings are available at runtime, the
+        docstrings of the attributes are not available. Instead of relying on
+        runtime information we parse the source code to extract the docstrings.
+        """
+        docs: dict[str, dict[str, str]] = {}
+        for file_path in self.iter(path):
+            if file_path.suffix != ".py":
+                continue
+            with open(file_path) as f:
+                src = f.read()
+                t = ast.parse(src)
+                docs.update(self.find_docstrings_in_module(t))
+        return docs
+
+    def find_docstrings_in_module(self, mod: ast.Module) -> dict[str, dict[str, str]]:
+        """
+        Find the docstrings for all the class attributes in the module. The
+        return dict has an entry for each class name, with a dict of attribute
+        name to docstring.
+
+        We look for all the (top level) class definitions in the module, and for
+        each class we look for attribute assignments with type annotations. If
+        we find such an assignment, we look for a string constant right after
+        the assignment, which we assume is the docstring for the attribute.
+        """
+        docs: dict[str, dict[str, str]] = {}
+        for stmt in mod.body:
+            if isinstance(stmt, ast.ClassDef):
+                class_name = stmt.name
+                docs[class_name] = {}
+                it = iter(stmt.body)
+                while True:
+                    try:
+                        node = next(it)
+                        # Look for an assignment with a type annotation
+                        if isinstance(node, ast.AnnAssign):
```

Comment:
Just curious, this seems like we look for assignments in any method of the class if I'm understanding it correctly.  Do people use docstrings outside of the constructor?  Or is it just more convenient to also include methods other than the constructor here?

Or is this only for class members?  Would e.g. https://github.com/pulumi/pulumi/pull/18540/files#diff-3988c9e539cf46802743600639fcf7cee5b63f141691642e07c75ee1892efe81R123-R124 also work as a docstring that's handled by this?

Ah and looking at the tests below, it looks like this works for class members as well, e.g.

```
    complex_output: pulumi.Output[ComplexType]
    """complex_output doc string"""
```

Naively I would have assumed that this is not an assignment, but if it is that's great!

Reply:
We don't actually look inside methods, we only look at the top level of the class body, where in the following example line 2 is an assignment with an annotation.

```python
class MyClass:
    a: int
```

https://docs.python.org/3/library/ast.html#ast.AnnAssign

Reply:
Ah nice, thanks for the reference!